include/nvexec/stream/split.cuh

```diff
@@ -181,12 +172,11 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
           if constexpr (stream_sender<Sender>) {
             STDEXEC_DBG_ERR(cudaEventDestroy(event_));
           }
-          STDEXEC_DBG_ERR(cudaStreamDestroy(stream_));
         }
       }
 
       env_t make_env() const noexcept {
-        return _split::__make_env(stop_source_, stream_);
+        return _split::__make_env(stop_source_, &const_cast<stream_provider_t&>(stream_provider_));
```

Comment:
I wonder if there's a way to keep this `const`-correct. What mutating operation does the env need of the stream provider?

Reply:
Env itself doesn't mutate the stream provider. It only distributes the reference to it. On the usage side, we need to register temporary storage destruction, which mutates the vector inside the stream provider.